pkg/cli/term/reader_windows.go

```diff
@@ -134,9 +134,13 @@ func convertEvent(event sys.InputEvent) Event {
 			// If only the shift is held down, we try and see if this is a
 			// non-functional key by looking if the rune generated is a
 			// printable ASCII character.
-			if 0x20 <= r && r < 0x7f {
+			if 0x20 <= r && r != 0x7f {
 				return KeyEvent(ui.Key{Rune: r})
 			}
+		} else if filteredMod&(leftCtrl|rightAlt) == leftCtrl|rightAlt && 0x20 <= r && r != 0x7f && currentKeyboardLayoutHasAltGr {
+			// Handle AltGr key combinations if they result in a rune
+			// Shift is also ignored, since it is required for some chars
+			return KeyEvent(ui.Key{Rune: r})
```

Comment:
There are also chars that require shift to be pressed, for example `ẞ` (uppercase ß), which won't be input if I use `filteredMod & ^(leftCtrl|rightAlt)`.

Additionally, I noticed that pressing `AltGr + A` results in `a` being input, which also isn't exactly right.

Reply:
> There are also chars that require shift to be pressed, for example ẞ (uppercase ß), which won't be input if I use filteredMod & ^(leftCtrl|rightAlt).

I see, that's quite a bit of finger gymnastics :) But I guess no character would require Ctrl or Alt to input? You can still use `filteredMod & ^(leftCtrl|rightAlt|shift)` as the mod instead of leaving it unset.

> Additionally, I noticed that pressing AltGr + A results in a being input, which also isn't exactly right (the A key has no alternate graving, so this should not create any character input).

Which terminal are you using?

I noticed the same behavior when testing in VS Code's integrated terminal, it's an event identical to a simple press on the `A` key, so there's nothing Elvish can do here - `cmd` also insert `a` when pressing `AltGr-A`.

However, neither Windows's default console nor the Windows Terminal app exhibits this behavior for me. Presumably VS Code does something funny with the keyboard events?

Reply:
I can confirm that the default console doesn't exhibit that behavior, but the VSCode console and wezterm do. I'll just use `filteredMod & ^(leftCtrl|rightAlt|shift)` and leave it at that.